src/raft/tests.rs

```diff
@@ -940,12 +1055,99 @@ async fn snapshot_install_unreliable_crash_2d() {
     snap_common(false, false, true).await;
 }
 
+/// do the servers persist the snapshots, and restart using snapshot along
+/// with the tail of the log?
+#[madsim::test]
+async fn snapshot_all_crash_2d() {
+    let servers = 3;
+    let iters = 5;
+
+    let t = RaftTester::new_with_snapshot(servers).await;
+    info!("Test (2D): crash and restart all servers");
+    let mut random = rand::rng();
+
+    t.one(random.gen_entry(), servers, true).await;
+
+    for _iter in 0..iters {
+        // enough to get a snapshot
+        for _ in 0..=SNAPSHOT_INTERVAL {
+            t.one(random.gen_entry(), servers, true).await;
+        }
+        let index1 = t.one(random.gen_entry(), servers, true).await;
+
+        // crash all
+        for i in 0..servers {
+            t.crash1(i);
+        }
+        // revive all
+        for i in 0..servers {
+            t.start1(i).await;
+            t.connect(i);
+        }
+
+        let index2 = t.one(random.gen_entry(), servers, true).await;
+        assert!(
+            index2 > index1,
```

Comment:
the go version say something like this
`if index2 < index1+1 {
			t.Fatalf("index decreased from %v to %v", index1, index2)
		}`